include/odp/api/spec/classification.h

```diff
@@ -164,9 +171,40 @@ typedef enum {
  * Used to communicate class of service creation options
  */
 typedef struct odp_cls_cos_param {
-	odp_queue_t queue;	/**< Queue associated with CoS */
-	odp_pool_t pool;	/**< Pool associated with CoS */
-	odp_cls_drop_t drop_policy;	/**< Drop policy associated with CoS */
+	/** Number of queues to be linked to this CoS.
+	 * If the number is greater than 1 then hashing is enabled.
+	 * If number is equal to 1 then hashing is disabled.
+	 * When hashing is enabled the queues are created by the implementation
+	 * and application need not configure any queue to the class of service.
+	 * When hashing is disabled application has to configure the queue to
+	 * the class of service.
+	 * Depening on the implementation this number might be rounded-off to
+	 * nearest supported value (e.g power of 2)
+	 */
+	uint32_t num_queue;
+
+	union {
+		/** Mapping used when num_queue = 1, hashing is disabled in
+		 * this case and application has to configure this queue and
+		 * packets are delivered to this queue */
+		odp_queue_t queue;
+
+		/** Mapping used when num_queue > 1, hashing is enabled in
+		 * this case and queues are created by the implementation */
+		struct {
+			/** Queue parameters */
+			odp_queue_param_t queue_param;
+
+			/** Protocol header fields which are included in
+			 * packet input hash calculation */
+			odp_pktin_hash_proto_t hash_proto;
+		};
+	};
+	/** Pool associated with CoS */
+	odp_pool_t pool;
+
+	/** Drop policy associated with CoS */
+	odp_cls_drop_t drop_policy;
 } odp_cls_cos_param_t;
```

Comment:
There are still doxygen issues with these API changes. Not sure why Travis isn't picking this up. 

```
bill@Ubuntu64:~/linaro/balacos$ make doxygen-doc
  DXGEN  doc/application-api-guide/Doxyfile
/home/bill/linaro/balacos/include/odp/api/spec/classification.h:202: warning: parameters of member odp_cls_cos_param::@3 are not (all) documented
  DXGEN  doc/helper-guide/Doxyfile
  DXGEN  doc/platform-api-guide/Doxyfile
/home/bill/linaro/balacos/include/odp/api/spec/classification.h:202: warning: parameters of member odp_cls_cos_param::@3 are not (all) documented
  DXGEN  doc/driver-api-guide/Doxyfile
bill@Ubuntu64:~/linaro/balacos$ 
```